src/index.ts

```diff
@@ -60,85 +64,244 @@ export interface DynamoDBAutoIncrementProps {
  * ```
  */
 export class DynamoDBAutoIncrement {
-  constructor(readonly props: DynamoDBAutoIncrementProps) {}
-
+  constructor(readonly props: DynamoDBAutoIncrementProps) {
+    this.#attributeName =
+      this.props.counterTableAttributeName ?? this.props.tableAttributeName
+  }
+  #attributeName: string
   async getLast(): Promise<number | undefined> {
     return (
       (
         await this.props.doc.get({
-          AttributesToGet: [this.props.counterTableAttributeName],
+          AttributesToGet: [this.#attributeName],
           Key: this.props.counterTableKey,
           TableName: this.props.counterTableName,
         })
-      ).Item?.[this.props.counterTableAttributeName] ?? undefined
+      ).Item?.[this.#attributeName] ?? undefined
+    )
+  }
+
+  async getLastItem() {
+    return (
+      (
+        await this.props.doc.get({
+          TableName: this.props.tableName,
+          Key: this.props.counterTableKey,
+        })
+      ).Item ?? undefined
     )
   }
 
   async put(item: Record<string, NativeAttributeValue>) {
-    for (;;) {
-      const counter = await this.getLast()
+    let nextCounter: number
+    let Update: UpdateCommandInput & { UpdateExpression: string }
 
-      let nextCounter: number
-      let Update: UpdateCommandInput & { UpdateExpression: string }
+    if (this.props.counterTableCopyItem) {
+      const ExpressionAttributeNames = this.#getExpressionAttributeNames(item)
+      const ExpressionAttributeValues = this.#getExpressionAttributeValues(item)
+      const UpdateExpression = this.#getUpdateExpression(item)
+      for (;;) {
+        const previousItem = await this.getLastItem()
+        const counter = previousItem?.[this.#attributeName]
```

Comment:
Why the new method? What is wrong with `this.getLast()`?

Reply:
`getLast` returns the latest value of the `#attributeName` from the table defined as `counterTableName`

`getLastItem` returns the latest value of the item defined on `tableName`, which is later PUT into the table defined as `counterTableName` 

The example case of circulars: 
  - `getLast` will return the latest circularId
  - `getLastItem` returns the latest circular